app/src/main/java/com/mashup/lastgarden/ui/scent/comment/ScentCommentBottomSheetFragment.kt

```diff
@@ -1,30 +1,63 @@
 package com.mashup.lastgarden.ui.scent.comment
 
+import android.app.Dialog
 import android.os.Bundle
 import android.view.LayoutInflater
+import android.view.MotionEvent
 import android.view.View
 import android.view.ViewGroup
 import androidx.core.view.isVisible
-import androidx.core.view.updateLayoutParams
 import androidx.fragment.app.activityViewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.paging.PagingData
+import androidx.recyclerview.widget.RecyclerView
 import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
 import com.google.android.material.bottomsheet.BottomSheetDialogFragment
 import com.mashup.base.autoCleared
-import com.mashup.base.utils.dp
+import com.mashup.base.extensions.hideSoftInput
+import com.mashup.base.image.GlideRequests
 import com.mashup.lastgarden.R
 import com.mashup.lastgarden.data.vo.Comment
 import com.mashup.lastgarden.databinding.ScentCommentBottomSheetBinding
 import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.flow.collectLatest
+import javax.inject.Inject
 
 @AndroidEntryPoint
 class ScentCommentBottomSheetFragment : BottomSheetDialogFragment(),
-    ScentCommentAdapter.OnClickListener {
+    ScentCommentPagingAdapter.OnClickListener {
     private var binding by autoCleared<ScentCommentBottomSheetBinding>()
     private val viewModel: ScentCommentViewModel by activityViewModels()
 
+    private lateinit var scentCommentAdapter: ScentCommentPagingAdapter
+
+    @Inject
+    lateinit var glideRequests: GlideRequests
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setStyle(STYLE_NORMAL, R.style.AppBottomSheetDialogTheme)
+        scentCommentAdapter = ScentCommentPagingAdapter(glideRequests, this)
+        val storyId = arguments?.getInt("storyId")
+        viewModel.setStoryId(storyId ?: 0)
+        viewModel.getCommentList(storyId ?: 0)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        dialog.setOnShowListener { dialogInterface ->
+            val bottomSheetDialog = dialogInterface as BottomSheetDialog
+            setupBottomSheet(bottomSheetDialog)
+        }
+        return dialog
+    }
+
+    private fun setupBottomSheet(bottomSheetDialog: BottomSheetDialog) {
+        val bottomSheet = bottomSheetDialog.findViewById<View>(R.id.commentBottomSheet) as View
```

Comment:
여기 as가 필요한가요??

Reply:
as View를 써주지 않으면 바로 밑의 코드 from(bottomSheet)에서 Type mismatch가 납니다..😢

Reply:
```suggestion
     private fun BottomSheetDialog.setupBottomSheet() {
```
이렇게 바꾸면 bottomSheetDialog를 넘기지 않고 확장함수로 사용할 수 있어요